CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+# Contributing to Thyra
+
+Thank you for considering contributing to Thyra! We welcome contributions from the community and value the time and effort you put into making this project better.
+
+## Getting Started
+
+To contribute to Thyra, you should have a basic understanding of the Go programming language and Git version control system. If you're new to Go, we recommend checking out [A Tour of Go](https://tour.golang.org/welcome/1) to get started.
+
+Before you can start contributing, you'll need to complete the following steps:
+- install the dependencies :
+```bash
+  sudo apt update
+  sudo apt install -y build-essential libgl1-mesa-dev xorg-dev p7zip
+```
+
+- Install Go: Go is required to build and run Thyra. You can install Go by following the instructions at [https://go.dev/doc/install](https://go.dev/doc/install).
+
+- Install Swagger: Thyra uses Swagger to generate code from the API documentation. You can install Swagger by running the following command:
+
+```bash
+go install github.com/go-swagger/go-swagger/cmd/swagger@latest
+```
+
+You can find more information about Swagger at [https://github.com/go-swagger/go-swagger](https://github.com/go-swagger/go-swagger).
+
+- Build generated files: Thyra generates code using Stringer and Go Swagger. You can build all generated files by running the following command:
+
+```bash 
+go generate ./web/...
+```
+
+- Run from source: Once you've completed the above steps, you can run Thyra from source by running the following commands:
+
+```bash
+go build -o thyra-server cmd/thyra-server/main.go
+sudo setcap CAP_NET_BIND_SERVICE=+eip thyra-server
+./thyra-server
+```
+## Code Formatting
+
+We take code formatting seriously in Thyra to maintain a consistent code style. Please follow these guidelines to ensure that your code is properly formatted:
+
+### Setup Code Formatting Tool
+
+We use Prettier to format our code. We recommend installing the Prettier extension for your IDE to format on save.
+
+For Go code, we use the formatter included in https://marketplace.visualstudio.com/items?itemName=golang.go.
+
+### golangci-lint
+
+We use `golangci-lint` to run linters in parallel. We recommend installing it locally and running it on your source code before pushing any modifications, otherwise some potential lint errors will be caught by the pipeline.
```

Comment:
```suggestion
We use `golangci-lint` to run linters in parallel. We recommend installing it locally and running it on your source code before pushing any modifications, otherwise some potential lint errors will be caught by the CI pipeline.
```